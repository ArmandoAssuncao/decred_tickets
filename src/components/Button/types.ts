export interface IProps {
    m: string
    onPress: () => void
    title: string
}

export interface IState {
}
