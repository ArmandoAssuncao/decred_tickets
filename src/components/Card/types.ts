import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    type: ITicket['status']
    value: number
}

export interface IState {
}
