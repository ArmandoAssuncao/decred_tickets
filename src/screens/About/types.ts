import { IScreenProps } from 'containers/Root/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
}
