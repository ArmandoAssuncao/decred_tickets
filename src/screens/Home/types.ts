import { IScreenProps } from 'containers/Root/types'
import { getTicket } from 'store/ticket/actions'
import { ITicketState } from 'store/ticket/types'

export interface IConnectedProps {
    ticket?: ITicketState
}

export interface IActionCreators {
    getTicket: typeof getTicket
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    stakePool: ITicketState['stakePool']
    ticketAddress: ITicketState['address']
}
