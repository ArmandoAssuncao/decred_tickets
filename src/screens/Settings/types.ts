import { IScreenProps } from 'containers/Root/types'
import { permitNotifications } from 'store/settings/actions'
import { ISettingsState } from 'store/settings/types'

export interface IConnectedProps {
    settings?: ISettingsState
}

export interface IActionCreators {
    permitNotifications: typeof permitNotifications
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    isNotifications: boolean
}
