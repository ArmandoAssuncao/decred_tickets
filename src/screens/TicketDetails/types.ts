import { IScreenProps } from 'containers/Root/types'
import { ITicket } from 'store/ticket/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps<INavigationState>, IConnectedProps, IActionCreators {
}

export interface IState {
}

export interface INavigationState {
    params: {
        tickets: ITicket[]
    }
}
