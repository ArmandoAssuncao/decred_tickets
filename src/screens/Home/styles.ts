import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    $borderRadius: 10,
    container: {
        backgroundColor: '$primaryColor',
        flex: 1,
    } as ViewStyle,
    scrollContentContainer: {
        padding: '0.7rem',
    } as ViewStyle,
    cardsContainer: {
        flexDirection: 'row',
    } as ViewStyle,
    input: {
        backgroundColor: '$secondColor',
        borderColor: '$secondColor',
        borderRadius: '$borderRadius',
        borderWidth: 1,
        flex: 1,
        fontSize: '1.2rem',
        paddingHorizontal: 10,
        textAlignVertical: 'top',
    },
    inputContainer: {
        flexDirection: 'row',
        marginTop: '1rem',
    } as ViewStyle,
    formContainer: {
        marginVertical: '0.8rem',
    } as ViewStyle,
    button: {
        marginLeft: '0.3rem',
    } as ViewStyle,
    pickerContainer: {
        flex: 1,
        flexDirection: 'row',
    } as ViewStyle,
})
