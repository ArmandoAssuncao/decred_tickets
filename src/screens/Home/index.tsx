import React from 'react'
import { ScrollView, TextInput, View } from 'react-native'
import { NavigationScreenConfig } from 'react-navigation'
import EStyleSheet from 'react-native-extended-stylesheet'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { IAppState } from 'store/types'

import i18n from 'config/i18n'
import { getTicket } from 'store/ticket/actions'
import { ITicketState, ITicket } from 'store/ticket/types'

import { IActionCreators, IConnectedProps, IProps, IState, } from './types'
import styles from './styles'

import { Button, Card, PickerStakePool } from 'components'

const mapStateToProps = (state: IAppState): IConnectedProps =>
  ({
    ticket: state.ticket,
  })

const mapDispatchToProps = (dispatch: any): MapDispatchToProps<IActionCreators, IProps> =>
    bindActionCreators(
        {
            getTicket,
        },
        dispatch
    )

export class Home extends React.PureComponent<IProps, IState> {
    static navigationOptions: NavigationScreenConfig<any> = {
        title: i18n.t('screens.home.title'),
    }

    constructor(props: IProps) {
        super(props)

        this.state = {
            stakePool: props.ticket.stakePool,
            ticketAddress: props.ticket.address,
        }
    }

    componentDidMount() {
        const { ticket } = this.props
        this.validAndGetTicket(ticket.address, ticket.stakePool)
    }

    private treatTicket(tickets: ITicket[]): any {
        const ticketsExpired = tickets.filter((ticket) => ticket.status === 'expired')
        const ticketsImmature = tickets.filter((ticket) => ticket.status === 'immature')
        const ticketsLive = tickets.filter((ticket) => ticket.status === 'live')
        const ticketsMissed = tickets.filter((ticket) => ticket.status === 'missed')
        const ticketsRevoked = tickets.filter((ticket) => ticket.status === 'revoked')
        const ticketsUnmined = tickets.filter((ticket) => ticket.status === 'unmined')
        const ticketsVoted = tickets.filter((ticket) => ticket.status === 'voted')

        return {
            expireds: ticketsExpired,
            immatures: ticketsImmature,
            lives: ticketsLive,
            misseds: ticketsMissed,
            revokeds: ticketsRevoked,
            unmineds: ticketsUnmined,
            voteds: ticketsVoted,
        }
    }

    private validAndGetTicket = (address: string, stakePool: ITicketState['stakePool']): void => {
        if (address && stakePool) {
            this.props.getTicket(address, stakePool)
        }
    }

    private onChangePicker = (stakePool: ITicketState['stakePool']): void => {
        this.setState({ stakePool })
    }

    private onChangeInputValue = (ticketAddress: string): void => {
        this.setState({ ticketAddress })
    }

    private onPressButton = (): void => {
        const { stakePool, ticketAddress } = this.state
        this.validAndGetTicket(ticketAddress, stakePool)
    }

    render(): JSX.Element {
        const { stakePool, ticketAddress } = this.state
        const tickets = this.treatTicket(this.props.ticket.tickets)

        return (
            <ScrollView style={styles.container}>
                <View style={styles.formContainer}>
                    <View style={styles.pickerContainer}>
                        <PickerStakePool value={stakePool} onChange={this.onChangePicker} />
                    </View>
                    <View style={styles.inputContainer}>
                        <TextInput
                            style={styles.input}
                            multiline={true}
                            underlineColorAndroid="transparent"
                            autoCapitalize="none"
                            autoCorrect={false}
                            blurOnSubmit={true}
                            numberOfLines={2}
                            placeholder={i18n.t('screens.home.input.placeholder')}
                            placeholderTextColor={EStyleSheet.value('$placeholderColor')}
                            returnKeyType="done"
                            selectTextOnFocus={true}
                            onChangeText={this.onChangeInputValue}
                            value={ticketAddress}
                        />
                        <Button
                            m={'OK'}
                            onPress={this.onPressButton}
                            title={i18n.t('screens.home.btn')}
                        />
                    </View>
                </View>
                <View style={styles.cardsContainer}>
                    <Card type={'revoked'} value={tickets.revokeds.length} />
                    <Card type={'voted'} value={tickets.voteds.length} />
                </View>
                <View style={styles.cardsContainer}>
                    <Card type={'expired'} value={tickets.expireds.length} />
                    <Card type={'missed'} value={tickets.misseds.length} />
                </View>
                <View style={styles.cardsContainer}>
                    <Card type={'unmined'} value={tickets.unmineds.length} />
                    <Card type={'immature'} value={tickets.immatures.length} />
                </View>
                <View style={styles.cardsContainer}>
                    <Card type={'live'} value={tickets.lives.length} />
                </View>
            </ScrollView>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
