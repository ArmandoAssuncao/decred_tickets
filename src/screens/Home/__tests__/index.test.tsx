import * as React from 'react'
import { shallowStore } from 'test/enzyme'

import Home from '../index'
import { ITicketState } from 'store/ticket/types'

describe('Home test render', () => {
    const store = {
        ticket: {
            address: 'address',
            error: false,
            stakePool: 'decredBrasil',
            tickets: [],
        } as ITicketState,
    }
    const wrapper = shallowStore(<Home getTicket={jest.fn()} />, store)

    it('Testing snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Testing instance of component', () => {
        expect(wrapper.instance()).toBeInstanceOf(Home.WrappedComponent)
    })
})
