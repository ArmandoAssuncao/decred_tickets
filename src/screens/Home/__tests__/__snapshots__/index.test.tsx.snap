// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test render Testing snapshot 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#0C1E3E",
      "flex": 1,
      "padding": 11.2,
    }
  }
>
  <Component
    style={
      Object {
        "marginVertical": 12.8,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <PickerStakePool
        onChange={[Function]}
        value="decredBrasil"
      />
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 16,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        multiline={true}
        numberOfLines={2}
        onChangeText={[Function]}
        placeholder="Ticket Address"
        placeholderTextColor="gray"
        returnKeyType="done"
        selectTextOnFocus={true}
        style={
          Object {
            "backgroundColor": "#F3F6F6",
            "borderColor": "#F3F6F6",
            "borderRadius": 10,
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 19.2,
            "paddingHorizontal": 10,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="transparent"
        value="address"
      />
      <Card
        m="OK"
        onPress={[Function]}
        title="OK"
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Card
      type="revoked"
      value={0}
    />
    <Card
      type="voted"
      value={0}
    />
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Card
      type="expired"
      value={0}
    />
    <Card
      type="missed"
      value={0}
    />
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Card
      type="unmined"
      value={0}
    />
    <Card
      type="immature"
      value={0}
    />
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Card
      type="live"
      value={0}
    />
  </Component>
</ScrollView>
`;
